From: Christian Kastner <ckk@kvr.at>
Date: Mon, 21 Dec 2015 13:57:20 +0000
Subject: Adjust tests for 3.18 kernel change

Backport a fix from upstream which addresses a change in the maximum
description size (4095 plus a NULL) in the upstream kernel from v3.18
onwards.

Origin: backport, https://git.kernel.org/cgit/linux/kernel/git/dhowells/keyutils.git/commit/?id=806621f9f3e5b56cd85d9972fd1348051a9911c8
Last-Update: 2016-05-01
---
 tests/toolbox.inc.sh | 19 ++++++++++++++-----
 1 file changed, 14 insertions(+), 5 deletions(-)

diff --git a/tests/toolbox.inc.sh b/tests/toolbox.inc.sh
index 16901a1..7647745 100644
--- a/tests/toolbox.inc.sh
+++ b/tests/toolbox.inc.sh
@@ -14,6 +14,8 @@
 includes=${BASH_SOURCE[0]}
 includes=${includes%/*}/
 
+. $includes/version.inc.sh
+
 echo === $OUTPUTFILE ===
 
 endian=`file -L /proc/$$/exe`
@@ -33,9 +35,18 @@ maxtypelen=31
 maxtype=`for ((i=0; i<$((maxtypelen)); i++)); do echo -n a; done`
 
 PAGE_SIZE=`getconf PAGESIZE`
-maxdesclen=$((PAGE_SIZE - 1))
-maxdesc=`for ((i=0; i<$((maxdesclen)); i++)); do echo -n a; done`
-maxcall=$maxdesc
+pagelen=$((PAGE_SIZE - 1))
+fullpage=`for ((i=0; i<$((pagelen)); i++)); do echo -n a; done`
+string4095=`for ((i=0; i<4095; i++)); do echo -n a; done`
+
+if ! version_less_than `uname -r` 3.18
+then
+    maxdesc=$string4095
+else
+    maxdesc=$fullpage
+fi
+
+maxcall=$fullpage
 
 maxsquota=`grep '^ *0': /proc/key-users | sed s@.*/@@`
 
@@ -84,8 +95,6 @@ function toolbox_report_result()
     fi
 }
 
-. $includes/version.inc.sh
-
 ###############################################################################
 #
 # Return true if the keyutils package being tested is older than the given
